### PR TITLE
Support call expressions

### PR DESCRIPTION
**Pull request by MikePopoloski**
- Bind and evaluate function call expressions and function statement bodies.
- Allocate storage for local variable definitions
- Handle built-in system function calls
- Add tests for the above functionality

### CODE REVIEW

File: tests/unit_tests/EvalTests.cpp [hunk 1]
```diff
@@ -15,4 +15,12 @@ TEST_CASE("Simple eval", "[eval") {
     CHECK(value.integer() == 13);
 }
 
+TEST_CASE("Function call", "[eval") {
```
Comment (drsam94) [inline tests/unit_tests/EvalTests.cpp]: typo, [eval]

also should also probably have a test for statement lists ?

Comment (MikePopoloski) [inline tests/unit_tests/EvalTests.cpp]: Yeah, but right now I only have support for return statements so there's not really anything to test there yet.

Comment (drsam94) [inline tests/unit_tests/EvalTests.cpp]: well, there is at least one list of statements you can test that I am not sure if is being handled properly now...

~~~
return 2;
return 1;
~~~

Comment (MikePopoloski) [inline tests/unit_tests/EvalTests.cpp]: True, I don't test for that and it actually doesn't work properly right now.

**Pull request merged**